cloudify_gcp/admin/projects.py

```diff
@@ -38,46 +48,66 @@ def get(self):
         details retrieval
         """
         self.logger.info('Get instance {0} details'.format(self.name))
-
         return self.discovery.projects().get(
             projectId=self.project_id).execute()
 
     @gcp.check_response
     def create(self):
-        project_body = {
-            constants.NAME: utils.get_gcp_resource_name(
-                ctx.node.properties[constants.NAME]),
-            'projectId': self.project_id
-        }
-        self.logger.info('Project info: {}'.format(repr(project_body)))
-        return self.discovery.projects().create(
-            body=project_body).execute()
+
+        try:
+            resource_exists = self.get()
+            return resource_exists
+        except Exception:
+            project_body = {
+                'name': self.name,
+                'projectId': self.project_id,
+                'parent': self.parent
+            }
+            self.logger.info('Project info: {}'.format(repr(project_body)))
+            result = self.discovery.projects().create(
+                body=project_body).execute()
+            return result
 
     @gcp.check_response
     def delete(self):
+        resource_exists = self.get()
+        if resource_exists.get('lifecycleState') != 'ACTIVE':
+            ctx.logger.info('The lifecycleState is not active, '
+                            'you must delete the project manually.'
+                            '{}'.format(resource_exists.get('lifecycleState')))
         return self.discovery.projects().delete(
             projectId=self.project_id).execute()
 
 
 @operation(resumable=True)
 @utils.throw_cloudify_exceptions
 def create(**_):
+
     if utils.resource_created(ctx, constants.RESOURCE_ID):
         return
 
     gcp_config = utils.get_gcp_config()
-
     project = Project(
-        gcp_config,
-        ctx.logger,
-        ctx.node.properties['id'],
-        ctx.node.properties[constants.NAME]
+        config=gcp_config,
+        logger=ctx.logger,
+        name=ctx.node.properties[constants.NAME],
+        project_id=ctx.node.properties.get('project_id', None),
+        parent=ctx.node.properties.get('parent', None)
     )
     utils.create(project)
```

Comment:
With the change to the project `get` and `create` functions, we can put some of the more procedural logic here in the operation:
```suggestion
resource_exists = project.get()
if not resource_exists:
    utils.create(project)
elif resource_exists.get('lifecycleStatus') == 'ACTIVE':
    return
raise OperationRetry('The project state is not ACTIVE yet.')
```